lib/makeup/lexers/elixir_lexer.ex

```diff
@@ -621,4 +650,21 @@ defmodule Makeup.Lexers.ElixirLexer do
     |> postprocess([])
     |> match_groups(group_prefix)
   end
+
+  @doc """
+  Register a custom lexer to be used for lexing sigil contents.
+
+  ## Examples
+
+      > Makeup.Lexers.ElixirLexer.register_sigil_lexer("H", Makeup.Lexers.HEExLexer)
+
+  """
+  def register_sigil_lexer(sigil, lexer) do
+    lexers = get_sigil_lexers()
+    Application.put_env(:makeup_elixir, :sigil_lexers, Map.put(lexers, sigil, lexer))
+  end
+
+  defp get_sigil_lexers() do
+    Application.get_env(:makeup_elixir, :sigil_lexers, %{})
```

Comment:
It is best to set the default in `def application` in `mix.exs` and here assume it has been set:

```suggestion
    Application.fetch_env!(:makeup_elixir, :sigil_lexers)
```